### DISCUSSION TITLE
turbo + vite: The sub package component library built by Vite, hot updated

### DISCUSSION THREAD

**Discussion by RSS1102** [vercel/turborepo#7951] [Help]
### Summary

In`packages/nordstjerna` I built a `solid` component library using `vite`and introduced `nordstjerna: workspace: "*"` in the `docs ` project created using `Astro`.

I want to change the code of the component library `nordstjerna` after run `turbo run dev`, so that docs can **hot update and change the page.**

### Example

https://github.com/RSS1102/nordstjerna

**Comment by anthonyshew** [comment]
This is how [the examples](https://turbo.build/repo/docs/getting-started/from-example) work. There may be something specific to your setup or tooling that isn't working/is misconfigured/doesn't support this. A Vite-based application should work, though, in my experience.

**Reply by RSS1102** [reply]
This is using Vite as a web server for HMR updates, but currently I want it to be used as a build tool to operate...
I am very foolish and don't know how to operate.

I also think my configuration is incorrect, so I tried many methods but they didn't work. I modified the[` 'exports':{
".": "./src/index. tsx"
}`](https://github.com/RSS1102/nordstjerna/blob/main/packages/nordstjerna/package.json#L8) I can achieve my goal, but I'm not sure if this will affect my use of publicly available packages to NPM.

---
maybe I need this feature for better [`watch mode `](https://github.com/vercel/turbo/issues/986). Can all files under a certain file be detected for changes in order to perform HMR updates.I don't know how the plan is going?
